libobs-bootstrapper/src/error.rs

```diff
@@ -1,18 +1,42 @@
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Debug)]
 pub enum ObsBootstrapError {
     GeneralError(String),
-    DownloadError(String),
+    InvalidFormatError(String),
+    /// Contains context and specific reqwest error
+    DownloadError(&'static str, reqwest::Error),
     ExtractError(String),
+    IoError(String),
```

Comment:
Should include the original IO error here and treat the string as explicit context for that error